Things_You_Should_Know/things_you_should_know.md

```diff
@@ -0,0 +1,400 @@
+
+|**CODE**|**CONTAINER**|
+|:------:|:-----------:|
+| ✅    | ✅          |
+
+If you're a *Node.js* developer, you can remotely debug your App using VS Code. You can set breakpoints, step through code, etc. on your App running in **App Service** from your development machine. For all of the details on how to remotely debug, check out [this blog post](https://medium.com/@auchenberg/introducing-remote-debugging-of-node-js-apps-on-azure-app-service-from-vs-code-in-public-preview-9b8d83a6e1f0) from Kenneth Auchenberg on the VS Code team.
+
+## Best Practices
+
+### You shouldn't install components using SSH because they won't persist
+
+|**CODE**|**CONTAINER**|
+|:------:|:-----------:|
+| ✅    | ✅          |
+
+You can **SSH** into your app container and install components. However, when your app is restarted, anything that you install will be gone. Why is that? That's the way Docker works. When your app starts, we run your Docker image and create a Docker container. Your app runs inside of that container, and the file system contains only what is in the Docker image. If your Docker image doesn't install a particular component, it won't be there when your container starts.
+
+If you want to install a particular component, make sure that you do it in a Dockerfile so that the component is included in your Docker image. See [this documentation](https://docs.microsoft.com/azure/app-service/containers/tutorial-custom-docker-image) for full information on how to do that.
+
+### The Bash console in Kudu (Advanced Tools) is a different container
+
+|**CODE**|**CONTAINER**|
+|:------:|:-----------:|
+| ✅    | ✅          |
+
+We offer a **BASH** console in [Kudu (Advanced Tools)](https://github.com/projectkudu/kudu/wiki) that you can use to browse directory structure, etc. It's important to realize that this console is running inside of **Kudu**. This is not the same container that your app is running in, so if you look at files and directories here, they won't reflect your app. The only exception to that is the /home directory. That directory is, by default, mounted from **App Service** built in storage, and it's the same location in both the **Kudu** and your app's container. [Learn more about persistent storage](#You-can-enable-and-disable-storage-persistence-with-an-app-setting)
+
+### You don't need (and should not have) SSL/TLS support in your app
+
+|**CODE**|**CONTAINER**|
+|:------:|:-----------:|
+| ✅    | ✅          |
+
+**App Service** terminates SSL/TLS at the front-ends. That means that SSL/TLS requests never get to your app. That's good news for you because it means that you don't need to (and should not) implement any support for SSL/TLS into your app. Also as stated above, it's important to understand that the front-ends where SSL/TLS is terminated are inside of our Azure data centers. If you use SSL/TLS with your app, your traffic across the Internet will always be safely encrypted.
```

Comment:
Confusing wording:  "If you use SSL/TLS in your app to make outbound calls,...."